### PR TITLE
Switch to OpenStreetMaps

### PR DESCRIPTION
**Pull request by gsvr**
Should get location from OpenStreetMaps Api using v 0.2 of LocationState.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -282,22 +285,70 @@ go.app = function() {
 
         self.states.add('state_suburb', function(name) {
             return new LocationState(name, {
+                map_provider: new OpenStreetMap({
```
Comment (gsvr) [inline src/app.js:288]: There's some hardcoded non-dry code here that will have to change in any case with the next release of `js-box-location` that should include landing of `https://github.com/praekelt/go-jsbox-location/issues/27`.

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: src/app.js [hunk 2]
```diff
@@ -282,22 +285,70 @@ go.app = function() {
...
+                    bounding_box: ["16.4500", "-22.1278", "32.8917", "-34.8333"],
```
Comment (imsickofmaps) [inline src/app.js:289]: add to `config`?

File: src/app.js [hunk 3]
```diff
@@ -282,22 +285,70 @@ go.app = function() {
...
+                    extract_address_data: function(result) {
+                        var formatted_address;
+                        if (!result.address) {
+                            formatted_address = result.display_name;
+                        } else {
+                            var city_town_village = result.address.city ||
+                                result.address.town || result.address.village;
+                            result.address.city_town_village = city_town_village;
+
+                            var addr_details = ['road', 'city_town_village',
```
Comment (imsickofmaps) [inline src/app.js]: drop `'postcode', 'country'` please

**Comment by imsickofmaps** [comment]
Looks good, just drop the extra fields and make the search on `town` and I think we're good

File: src/app.js (hunk 2 above)
Comment (gsvr) [inline src/app.js:289]: Opened a new ticket: https://github.com/praekelt/go-usaid-clinicfinder/issues/13

File: src/app.js (hunk 1 above)
Comment (gsvr) [inline src/app.js:288]: Opened a new ticket: https://github.com/praekelt/go-usaid-clinicfinder/issues/14

**Comment by imsickofmaps** [comment]
:+1:

**Comment by gsvr** [comment]
Opened a new issue: https://github.com/praekelt/go-jsbox-location/issues/30, and changed address limit to 4.

**Comment by imsickofmaps** [comment]
@gsvr Travis happy, I added a commit to the README to trigger a test run. :+1:

**Pull request merged**